tasks/ModelTask.cpp

```diff
@@ -1,198 +0,0 @@
-/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
-//======================================================================================
-// Brazilian Institute of Robotics 
-// Authors: Thomio Watanabe
-// Date: December 2014
-//====================================================================================== 
-// add joints, position and velocity
-//======================================================================================
-#include "ModelTask.hpp"
-//======================================================================================
-gazebo::ModelTask::ModelTask(std::string const& name, TaskCore::TaskState initial_state)
-	: ModelTaskBase(name, initial_state),environment(0)
-{
-}
-//======================================================================================
-gazebo::ModelTask::ModelTask(std::string const& name, RTT::ExecutionEngine* engine, TaskCore::TaskState initial_state)
-	: ModelTaskBase(name, engine, initial_state),environment(0)
-{
-}
-//======================================================================================
-gazebo::ModelTask::~ModelTask()
-{
-	delete joint_port;
-	delete link_port;
-	
-	for(JointPort_V::iterator it = joint_port_list.begin();it != joint_port_list.end(); ++it)
-		delete (it)->first;
-	
-	for(LinkPort_V::iterator it = link_port_list.begin(); it != link_port_list.end(); ++it)
-		delete it->first; 
-	
-	joint_port_list.clear();
-	link_port_list.clear();
-}
-//======================================================================================
-void gazebo::ModelTask::setGazeboModel(physics::WorldPtr _world,  physics::ModelPtr _model, int _environment)
-{
-	world = _world;
-	model = _model;
-	environment = _environment;
-//	sdf = model->GetSDF();
-	
-//	Export Gazebo Joints and Links to Rock-Robotics
-	setJoints();
-	setLinks();
-} 
-//======================================================================================
-void gazebo::ModelTask::setJoints()
-{
-	typedef gazebo::physics::Joint_V Joint_V;
-	Joint_V joints;
-			
-	// Get all joints from a model and creates a Rock component InputPort
-	joints = model->GetJoints();
-	for(Joint_V::iterator joint = joints.begin(); joint != joints.end(); ++joint)
-	{
-		gzmsg <<"RockBridge: found joint: "<< world->GetName() + "/" + model->GetName() + 
-				"/" + (*joint)->GetName() << std::endl;
-		gzmsg << "RockBridge: create joint InputPort in Rock." << std::endl;
-		joint_port = new RTT::InputPort<double>( "joint: " + world->GetName() +
```

Comment:
This is a red flag. You assign to a pointer that is not a local variable within a loop. Which means that (1) either you have a memory leak or (2) you have a potential problem with double-free. In your case, you get the double-free.

Define your variables in the most narrow scope possible: where you start using them. Whenever you read your code, always ask yourself when you assign to "permanent" variables (i.e. object attributes)